hathor/verification/transaction_verifier.py

```diff
@@ -235,21 +243,47 @@ def verify_sum(self, token_dict: dict[TokenUid, TokenInfo]) -> None:
                 # that's the usual behavior, nothing to do
                 pass
             elif token_info.amount < 0:
-                # tokens have been melted
-                if not token_info.can_melt:
-                    raise InputOutputMismatch('{} {} tokens melted, but there is no melt authority input'.format(
-                        token_info.amount, token_uid.hex()))
-                withdraw += get_withdraw_amount(self._settings, token_info.amount)
+                if token_info.version == TokenInfoVersion.FEE:
```

Comment:
Maybe you should use `match` here. This will allow linter to check whether you are covering all possibilities.